src/wp-includes/formatting.php

```diff
@@ -3914,10 +3914,26 @@ function wp_trim_excerpt( $text = '', $post = null ) {
 		$text = strip_shortcodes( $text );
 		$text = excerpt_remove_blocks( $text );
 
+		/*
+		 * Temporarily unhook wp_filter_content_tags() since any tags
+		 * within the excerpt are stripped out. Modifying the tags here
+		 * is wasteful and can lead to bugs in the image counting logic.
+		 */
+		$filter_removed = false;
+		if ( has_filter( 'the_content', 'wp_filter_content_tags' ) ) {
+			remove_filter( 'the_content', 'wp_filter_content_tags' );
+			$filter_removed = true;
+		}
```

Comment:
```suggestion
		$filter_removed = remove_filter( 'the_content', 'wp_filter_content_tags' );
```

Could we make this simpler by doing this? `remove_filter` returns a boolean.

Reply:
That's a great idea, will update to simplify.